src/ripple/app/ledger/LedgerEntrySet.cpp

```diff
@@ -1024,52 +1024,66 @@ uint256 LedgerEntrySet::getNextLedgerIndex (
 
 void LedgerEntrySet::incrementOwnerCount (SLE::ref sleAccount)
 {
+    increaseOwnerCount (sleAccount, 1);
+}
+
+void LedgerEntrySet::incrementOwnerCount (Account const& owner)
+{
+    increaseOwnerCount(
+        entryCache (ltACCOUNT_ROOT, getAccountRootIndex (owner)), 1);
+}
+
+void
+LedgerEntrySet::increaseOwnerCount (SLE::ref sleAccount, std::size_t howMuch)
+{
     assert (sleAccount);
 
     std::uint32_t const current_count = sleAccount->getFieldU32 (sfOwnerCount);
+    std::uint32_t new_count = current_count + howMuch;
 
-    if (current_count == std::numeric_limits<std::uint32_t>::max ())
+    // Check for integer overflow -- well defined behavior on unsigned.
+    if (new_count < current_count)
     {
         WriteLog (lsFATAL, LedgerEntrySet) <<
             "Account " << sleAccount->getFieldAccount160 (sfAccount) <<
             " owner count exceeds max!";
-        return;
+        new_count = std::numeric_limits<std::uint32_t>::max ();
     }
-
-    sleAccount->setFieldU32 (sfOwnerCount, current_count + 1);
+    sleAccount->setFieldU32 (sfOwnerCount, new_count);
     entryModify (sleAccount);
 }
 
-void LedgerEntrySet::incrementOwnerCount (Account const& owner)
+void LedgerEntrySet::decrementOwnerCount (SLE::ref sleAccount)
 {
-    incrementOwnerCount(entryCache (ltACCOUNT_ROOT,
-        getAccountRootIndex (owner)));
+    decreaseOwnerCount (sleAccount, 1);
 }
 
-void LedgerEntrySet::decrementOwnerCount (SLE::ref sleAccount)
+void LedgerEntrySet::decrementOwnerCount (Account const& owner)
+{
+    decreaseOwnerCount(
+        entryCache (ltACCOUNT_ROOT, getAccountRootIndex (owner)), 1);
+}
+
+void
+LedgerEntrySet::decreaseOwnerCount (SLE::ref sleAccount, std::size_t howMuch)
 {
     assert (sleAccount);
 
     std::uint32_t const current_count = sleAccount->getFieldU32 (sfOwnerCount);
+    std::uint32_t new_count = current_count - howMuch;
 
-    if (current_count == 0)
+    // Check for integer underflow -- well defined behavior on unsigned.
+    if (new_count > current_count)
     {
         WriteLog (lsFATAL, LedgerEntrySet) <<
             "Account " << sleAccount->getFieldAccount160 (sfAccount) <<
-            " owner count is already 0!";
-        return;
+            " owner count set below 0!";
+        new_count = 0;
```

Comment:
I'm more than happy to add an `assert` here.  Would that help much?  I'm under the impression that a debug rippled has trouble keeping up with the network traffic.  On the other hand it can't hurt, since the production servers are all running optimized code.  Thoughts?